### PR TITLE
adding view for following

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: users/serializers.py [hunk 1]
```diff
@@ -2,12 +2,23 @@
...
 
 
 class UserPublicSerializer(serializers.ModelSerializer):
     id = serializers.IntegerField(read_only=True)
     profile_photo = Base64ImageField()
+    relations = serializers.SerializerMethodField()
+
+    def get_relations(self, obj):
+        request = self.context.get("request")
+        if request:
```
Comment (revimi) [inline users/serializers.py]: May it be null? Do we need this check?

File: users/serializers.py [hunk 2]
```diff
@@ -2,12 +2,23 @@
...
 from rest_framework import serializers
 
-from users.models import User
+from users.models import Followers, User
 
 
 class UserPublicSerializer(serializers.ModelSerializer):
     id = serializers.IntegerField(read_only=True)
     profile_photo = Base64ImageField()
+    relations = serializers.SerializerMethodField()
```
Comment (revimi) [inline users/serializers.py:11]: What about having just `is_followed` field? Or you are going to add more relations here?

File: users/urls.py [hunk 3]
```diff
@@ -2,15 +2,24 @@
...
 
 from users.views import (
+    FollowerViewSet,
     RetrieveUpdateSelfUserProfileView,
     RetrieveUserProfileView,
     SearchUserView,
     SignupView,
 )
 
+set_follower = FollowerViewSet.as_view(
```
Comment (revimi) [inline users/urls.py]: Follower is someone you follow. Probably you mean `set-following` or `follow`

File: users/views.py [hunk 4]
```diff
@@ -39,3 +40,15 @@ class SearchUserView(generics.ListAPIView):
...
+class FollowerViewSet(viewsets.ModelViewSet):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, pk: int):
+        Followers.objects.update_or_create(follower=request.user, following_id=pk)
+        return Response(status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, pk: int) -> Response:
+        Followers.objects.filter(follower=request.user, following_id=pk).delete()
+        return Response(status=status.HTTP_200_OK)
```
Comment (revimi) [inline users/views.py:54]: 200 is ok here. An alternative for delete endpoint is `204 No Content`

File: users/serializers.py (hunk 1 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: Testing doesn't work without this check

File: users/serializers.py (hunk 2 above)
Comment (marinaRomanchuk) [reply users/serializers.py:11]: Maybe we can add the number of followers

File: users/views.py (hunk 4 above)
Comment (marinaRomanchuk) [reply users/views.py:54]: Done

File: users/urls.py (hunk 3 above)
Comment (marinaRomanchuk) [reply users/urls.py]: Yes, renamed

File: users/serializers.py (hunk 1 above)
Comment (revimi) [reply users/serializers.py]: The code shouldn't follow the tests. We need to add some fixture for dummy request in tests

File: users/serializers.py (hunk 2 above)
Comment (revimi) [reply users/serializers.py:11]: ok

File: users/views.py [hunk 5]
```diff
@@ -39,3 +40,18 @@ class SearchUserView(generics.ListAPIView):
...
+
+class FollowerViewSet(viewsets.ModelViewSet):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, pk: int):
+        Followers.objects.update_or_create(follower=request.user, following_id=pk)
+        return Response(status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, pk: int) -> Response:
+        to_delete = Followers.objects.filter(follower=request.user, following_id=pk)
```
Comment (revimi) [inline users/views.py]: I did not mean that we have 2 different cases. I mean that we can return 200 or 204, both are ok. So you can turn the previous version back

Comment (marinaRomanchuk) [reply users/views.py]: Done

File: users/serializers.py (hunk 1 above)
Comment (marinaRomanchuk) [reply users/serializers.py]: I see, by the way I've found what was the problem, so this check was removed and tests are ok

**Review by revimi — APPROVED** [review]

**Pull request merged**